src/themis/sym_enc_message.c

```diff
@@ -221,8 +221,23 @@ themis_status_t themis_auth_sym_decrypt_message(const uint8_t* key,
 						size_t* message_length){
   uint8_t key_[THEMIS_AUTH_SYM_KEY_LENGTH/8];
   THEMIS_CHECK_PARAM(context!=NULL && context_length!=0);
-  THEMIS_STATUS_CHECK(themis_sym_kdf(key,key_length, THEMIS_SYM_KDF_KEY_LABEL, (uint8_t*)(&encrypted_message_length), sizeof(encrypted_message_length), in_context, in_context_length, key_, sizeof(key_)),THEMIS_SUCCESS);
-  return themis_auth_sym_decrypt_message_(key_, sizeof(key_), in_context, in_context_length, context, context_length, encrypted_message, encrypted_message_length, message, message_length);
+  THEMIS_STATUS_CHECK(themis_sym_kdf(key,key_length, THEMIS_SYM_KDF_KEY_LABEL, (uint8_t*)(&encrypted_message_length), sizeof(uint32_t), in_context, in_context_length, key_, sizeof(key_)),THEMIS_SUCCESS);
+  themis_status_t decryption_result = themis_auth_sym_decrypt_message_(key_, sizeof(key_), in_context, in_context_length, context, context_length, encrypted_message, encrypted_message_length, message, message_length);
+
+#ifdef THEMIS_097_SECURE_CELL_X64_COMPATIBILITY_FIX
+
+  if (decryption_result != THEMIS_SUCCESS && decryption_result != THEMIS_BUFFER_TOO_SMALL) {
+
+    // we are on x64, should sizeof(uin64_t) for backwards compatibility with themis 0.9.6 x64
+    if (sizeof(size_t) == sizeof(uint64_t)) {
```

Comment:
you mean, the MACROS code may look smth like

```
#ifdef COMPAT

  #define THEMIS_SCELL_COMPAT(func){ \
    /* below will likely be evaluated compile time, so for other cases it will be a no-op */
    if (sizeof(size_t) == sizeof(uint64_t) { \
     // compatibility code
      {func;} \
    } \
  }

#else
  #define THEMIS_SCELL_COMPAT(func) {}
#endif
}
```

usage code

```
  if (decryption_result != THEMIS_SUCCESS && decryption_result != THEMIS_BUFFER_TOO_SMALL) {

    THEMIS_SCELL_COMPAT({
        THEMIS_STATUS_CHECK(themis_sym_kdf(key,key_length, THEMIS_SYM_KDF_KEY_LABEL, (uint8_t*)(&encrypted_message_length), sizeof(uint64_t), NULL, 0, key_, sizeof(key_)),THEMIS_SUCCESS);
        decryption_result = themis_sym_decrypt_message_u_(key_,sizeof(key_),context,context_length,encrypted_message,encrypted_message_length,message,message_length);
    });
  }
```

Reply:
Well I don't think we actually need `#define THEMIS_SCELL_COMPAT(func)` at all, just

```
#ifdef COMPAT
if (sizeof(size_t) == sizeof(uint64_t) { ...
}
#endif
```